src/MessagePack.GeneratorCore/Generator/FormatterTemplate.tt

```diff
@@ -90,7 +90,7 @@ namespace <#= Namespace #>
             global::MessagePack.IFormatterResolver formatterResolver = options.Resolver;
 <# } #>
             var length = reader.ReadArrayHeader();
-<# var canOverwrite = objInfo.ConstructorParameters.Length == 0;
+<# var canOverwrite = objInfo.ConstructorParameters.Length == 0 && !objInfo.HasInitOnlySetter;
```

Comment:
For the processing of temporary variables, reused the existing code of 'canOverrite' case.